src/webgpu/api/validation/render_pipeline/createRenderPipeline.spec.ts

```diff
@@ -220,8 +249,204 @@ g.test('color_formats_must_be_renderable')
+      .combine('dstFactor', kBlendFactors)
+      .combine('operation', kBlendOperations)
+  )
+  .fn(async t => {
+    const { isAsync, component, srcFactor, dstFactor, operation } = t.params;
+
+    const defaultBlendComponent: GPUBlendComponent = {
+      srcFactor: 'src-alpha',
+      dstFactor: 'dst-alpha',
+      operation: 'add',
+    };
+    const blendComponentToTest: GPUBlendComponent = {
+      srcFactor,
+      dstFactor,
+      operation,
+    };
+    const format = 'rgba8unorm';
+
+    const descriptor = t.getDescriptor({
+      targets: [
+        {
+          format,
+          blend: {
+            color: component === 'color' ? blendComponentToTest : defaultBlendComponent,
+            alpha: component === 'alpha' ? blendComponentToTest : defaultBlendComponent,
+          },
+        },
+      ],
+    });
+
+    if (operation === 'min' || operation === 'max') {
+      const _success = srcFactor === 'one' && dstFactor === 'one';
+      t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
+    } else {
+      t.doCreateRenderPipelineTest(isAsync, true, descriptor);
+    }
+  });
+
+g.test('fragment_state,targets_write_mask')
+  .desc(`Tests that color target state write mask must be < 16.`)
+  .params(u =>
+    u
+      .combine('isAsync', [false, true])
+      .combine('writeMask', [0, 0x1, 0x2, 0x7, 0x8, 0xf, 0x10, 0xff])
+  )
+  .fn(async t => {
+    const { isAsync, writeMask } = t.params;
+
+    const descriptor = t.getDescriptor({
+      targets: [
+        {
+          format: 'rgba8unorm',
+          writeMask,
+        },
+      ],
+    });
+
+    t.doCreateRenderPipelineTest(isAsync, writeMask < 16, descriptor);
+  });
+
+g.test('primitive_state,strip_index_format')
+  .desc(
+    `If primitive.topology is not "line-strip" or "triangle-strip", primitive.stripIndexFormat must be undefined.`
+  )
+  .params(u =>
+    u
+      .combine('isAsync', [false, true])
+      .combine('topology', [undefined, ...kPrimitiveTopology] as const)
+      .combine('stripIndexFormat', [undefined, ...kIndexFormat] as const)
+  )
+  .fn(async t => {
+    const { isAsync, topology, stripIndexFormat } = t.params;
+
+    const descriptor = t.getDescriptor({ primitive: { topology, stripIndexFormat } });
+
+    const _success =
+      topology === 'line-strip' || topology === 'triangle-strip' || stripIndexFormat === undefined;
+    t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
+  });
+
+g.test('primitive_state,unclipped_depth')
+  .desc(`If primitive.unclippedDepth is true, features must contain "depth-clip-control".`)
+  .params(u => u.combine('isAsync', [false, true]).combine('unclippedDepth', [false, true]))
+  .fn(async t => {
+    const { isAsync, unclippedDepth } = t.params;
+
+    const descriptor = t.getDescriptor({ primitive: { unclippedDepth } });
+
+    const _success = !unclippedDepth || t.device.features.has('depth-clip-control');
+    t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
+  });
```

Comment:
There's a redundant stub in `src/webgpu/api/validation/capability_checks/features/depth_clip_control.spec.ts` that can be deleted now